docker/1.11.0/Dockerfile.gpu

```diff
@@ -0,0 +1,89 @@
+    && rm -rf /var/lib/apt/lists/*
+
+ENV NCCL_VERSION=2.3.5-2+cuda9.0
+ENV CUDNN_VERSION=7.3.1.20-1+cuda9.0
+ENV TF_TENSORRT_VERSION=4.1.2
+
+RUN buildDeps=" \
+        ca-certificates \
+        cuda-command-line-tools-9-0 \
+        cuda-cublas-dev-9-0 \
+        cuda-cudart-dev-9-0 \
+        cuda-cufft-dev-9-0 \
+        cuda-curand-dev-9-0 \
+        cuda-cusolver-dev-9-0 \
+        cuda-cusparse-dev-9-0 \
+        curl \
+        libcudnn7=${CUDNN_VERSION} \
+        libnccl2=${NCCL_VERSION} \
+        libgomp1 \
+    " \
+    && apt-get update && apt-get install -y --no-install-recommends $buildDeps \
+    # The 'apt-get install' of nvinfer-runtime-trt-repo-ubuntu1604-4.0.1-ga-cuda9.0
+    # adds a new list which contains libnvinfer library, so it needs another
+    # 'apt-get update' to retrieve that list before it can actually install the
+    # library.
+    # We don't install libnvinfer-dev since we don't need to build against TensorRT,
+    # and libnvinfer4 doesn't contain libnvinfer.a static library.
+    && apt-get update && apt-get install -y --no-install-recommends \
+        nvinfer-runtime-trt-repo-ubuntu1604-4.0.1-ga-cuda9.0 \
+    && apt-get update && apt-get install -y --no-install-recommends \
+        libnvinfer4=${TF_TENSORRT_VERSION}-1+cuda9.0 \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm /usr/lib/x86_64-linux-gnu/libnvinfer_plugin* \
+    && rm /usr/lib/x86_64-linux-gnu/libnvcaffe_parser* \
+    && rm /usr/lib/x86_64-linux-gnu/libnvparsers* \
+    && if [ $py_version -eq 3 ]; \
+           then apt-get update && apt-get install -y --no-install-recommends python3.6 \
+                && ln -s -f /usr/bin/python3.6 /usr/bin/python; \
+           else apt-get update && apt-get install -y --no-install-recommends python; fi
+
+# Python won’t try to write .pyc or .pyo files on the import of source modules
+ENV PYTHONDONTWRITEBYTECODE=1 PYTHONUNBUFFERED=1
+
+RUN curl -fSsL -O https://bootstrap.pypa.io/get-pip.py && \
+    python get-pip.py \
+		    --disable-pip-version-check \
+		    --no-cache-dir \
+		    "pip==18.1" \
+	; \
+	pip --version; \
+	find /usr/local -depth \
+		\( \
+			\( -type d -a \( -name test -o -name tests \) \) \
+			-o \
+			\( -type f -a \( -name '*.pyc' -o -name '*.pyo' \) \) \
+		\) -exec rm -rf '{}' +; \
+    rm get-pip.py
+
+WORKDIR /root
+
+COPY $framework_installable .
+COPY $framework_support_installable .
+
+RUN framework_installable_local=$(basename $framework_installable) \
+    && framework_support_installable_local=$(basename $framework_support_installable) \
+    \
+    && pip install --no-cache --upgrade $framework_installable_local \
+    && pip install $framework_support_installable_local \
+    && pip install "sagemaker-tensorflow>=1.11,<1.12" \
+    \
+    && rm $framework_installable_local \
+    && rm $framework_support_installable_local
+
+ENV SAGEMAKER_TRAINING_MODULE sagemaker_tensorflow_container.training:main
```

Comment:
You can use a linter like https://github.com/hadolint to help you here.

Reply:
I have run this locally, only thing comes up now is the pin version stuff.  I will add it to the build in the next pr.